index.html

```diff
@@ -0,0 +1,475 @@
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
+                                      provides: "{ tripId }")
+}
+```
+
+The meaning of these directives is explored the [Directives](#directives) section.
+
+## Actors
+
+```mermaid
+#::[Actors and roles within an example composition pipeline]
+flowchart TB
+    subgraph A [subgraph A]
+      schemaA([schema A])
+      endpointA([endpoint A])
+    end    
+    subgraph B [subgraph B]
+      schemaB([schema B])
+      endpointB([endpoint B])
+    end
+    subgraph C [subgraph C]
+      schemaC([schema C])
+      endpointC([endpoint C])
+    end
+    subgraph Producer
+      Composer
+    end
+    CSDL([CSDL])
+    subgraph Consumer
+      Router
+    end    
+    A-->Composer
+    B-->Composer
+    C-->Composer
+    Composer-->CSDL
+    CSDL-->Router
+    Router-->published([Published Schema])
+    published-->Clients
+    Clients-->Router
+```
+
+<a name=def-producer></a>
+**Producers** generate CSDL. This spec places requirements on csdl producers.
+
+<a name=def-consumer></a>
+**Consumers** consume CSDL. This spec places requirements on csdl consumers.
+
+<a name=def-composer></a>
+**Composers** or **compilers** are producers which compose subgraph schemas into CSDL. This document places no particular requirements on the composition algorithm, except that it must produce valid csdl.
+
+<a namme=def-router></a>
+**Routers** are consumers which serve a composed schema as a GraphQL endpoint.
+  - Graph routers differ from standard GraphQL endpoints in that they are not expected to process data or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch.
+  - Routers **may**—and typically will—omit CSDL directives and types from the schema they present to clients via introspection
+
+<a name=def-endpoint></a>
+**Endpoints** are running servers which can resolve GraphQL queries against a schema. In this version of the spec, endpoints must be URLs, typically http/https URLs.
+
+<a name=def-subgraph></a>
+**Subgraphs** are the GraphQL schemas which were composed to form the CSDL. A subgraph has:
+  - a name, which must be unique within the CSDL
+  - an [endpoint](#endpoint)
+
+This spec does not place any requirements on subgraph schemas. Generally, they may be of any shape. In particular, subgraph schemas do not need to provide a CSL; neither is it an error for them to do so. Composers **may** place additional requirements on subgraph schemas to aid in composition; composers **should** document any such requirements.
+
+## Structure
+
+A CSDL document must be a valid [GraphQL schema definition language](https://spec.graphql.org/draft/#sec-Type-System) document.
```

Comment:
If we're linking to the spec we should probably use its terminology.

> A CSDL document must be a valid GraphQL Document consisting only of `TypeSystemDefinition`s (not `ExecutableDefinition`s or `TypeSystemExtension`s).

If we want to use the term SDL we could define it ourselves as short for schema definition language and as a GraphQL Document consisting of `TypeSystemDefinition`s and `TypeSystemExtension`s.

I think you also want to avoid the word "valid". This suggests GraphQL's "Validation" concept (section 5 of the spec), which I believe only works on Documents of `ExecutableDefinition`s. So maybe more like

> A CSDL document is a string (??) that parses as a GraphQL Document consisting only of `TypeSystemDefinition`s (not `ExecutableDefinition`s or `TypeSystemExtension`s).

(If we keep around the concept of the serialization at all, that is.)

Reply:
Would love it if we could define SDL here as David suggested, it'd be nice if we had a spec definition to point to in the future whenever we used the phrase "SDL" and this could be that place.

Regarding "valid"/"validation", the GraphQL spec indeed dedicates Section 5 to the validation of a GraphQL request in the context of a schema, and calls that section "Validation". But it's also worth noting that the sub-sections in Section 3 (which governs GraphQL type systems, a.k.a. schemas) often have a sub-sub-section called "Type Validation", which talks about validation checks for those types. I think it's okay to assume that when people see the word "valid" when talking about a schema, that they know we're talking about these type validation rules. To be explicit though, we could add a section to this spec that says a schema written in SDL is "valid" if it succeeds the validations in Section 3, which removes ambiguity.